src/typeahead/typeahead.js

```diff
@@ -44,6 +44,10 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.debounce', 'ui.bootstrap
 
     //should it restrict model values to the ones selected from the popup only?
     var isEditable = originalScope.$eval(attrs.typeaheadEditable) !== false;
+    originalScope.$watch(attrs.typeaheadEditable, function (newVal) {
+      isEditable = newVal !== false;
+      modelCtrl.$setValidity('editable', isEditable);
```

Comment:
This is a breaking change, and editable state should not be related to ngModel's validity. This is a bad change.